canvas_modules/common-canvas/src/palette/palette-dialog-content-categories.jsx

```diff
@@ -32,8 +32,9 @@ class PaletteDialogContentCategories extends React.Component {
 		for (var idx = 0; idx < this.props.categories.length; idx++) {
 			catDivs.push(
 				<PaletteDialogContentCategory
-					key={this.props.categories[idx]}
-					categoryName={this.props.categories[idx]}
+					key={this.props.categories[idx].label}
+					categoryName={this.props.categories[idx].label}
+					categoryId={this.props.categories[idx].id}
```

Comment:
Since we are accessing two properties from the category object I think I'd be tempted to just pass the category object into the React component and then just access the label and id as needed (as is done in the flyout palette category)